s3fs/tests/test_s3fs.py

```diff
@@ -267,6 +267,17 @@ def test_move(s3):
     assert not s3.exists(fn)
 
 
+def test_get_put(s3):
+    import tempfile
+    fn = tempfile.mktemp()
+    s3.get(test_bucket_name+'/test/accounts.1.json', fn)
+    data = files['test/accounts.1.json']
+    assert open(fn, 'rb').read() == data
+    s3.put(fn, test_bucket_name+'/temp')
+    assert s3.du(test_bucket_name+'/temp')[test_bucket_name+'/temp'] == len(data)
+    assert s3.cat(test_bucket_name+'/temp') == data
```

Comment:
I thought the OS took responsibility for clearing tempfiles from processes that end. If not, I may have many of these around...

tmpfile is a little more complex than I would have expected; anything wrong with the following:
```python
@ontextmanager
def tmpfile():
    fn = tempfile.mktmp()
    try:
        yield fn
    finally:
        if os.exists(fn):
            os.remove(fn)